### PR TITLE
Support nested property path

### PR DESCRIPTION
**Pull request by runceel**
#197

### CODE REVIEW

File: Source/ReactiveProperty.Core/Internals/AccessorCache.cs [hunk 1]
```diff
@@ -84,4 +128,138 @@ private static Delegate CreateSetAccessor<TProperty>(Expression<Func<TType, TPro
...
+                if (setCache.TryGetValue(propertyName, out var accessor))
+                {
+                    return accessor;
+                }
+
+                return CreateAndCacheSetAccessor(type, propertyName, setCache);
+            }
+        }
+
+        public static bool IsNotNestedPropertyPath<TSubject, TProperty>(Expression<Func<TSubject, TProperty>> propertySelector)
```
Comment (xin9le) [inline Source/ReactiveProperty.Core/Internals/AccessorCache.cs:213]: Should prefer a positive name like `IsNestedPropertyPath`.

Comment (runceel) [reply Source/ReactiveProperty.Core/Internals/AccessorCache.cs]: Fixed!

File: Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs [hunk 2]
```diff
@@ -37,8 +36,23 @@ public static IObservable<PropertyChangedEventArgs> PropertyChangedAsObservable<
             bool isPushCurrentValueAtFirst = true)
             where TSubject : INotifyPropertyChanged
         {
-            var accessor = AccessorCache<TSubject>.LookupGet(propertySelector, out var propertyName);
+            if (!ExpressionTreeUtils.IsNestedPropertyPath(propertySelector))
```
Comment (xin9le) [inline Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs:39]: It's easier to understand if swap `if` and `else`.

```cs
return ExpressionTreeUtils.IsNestedPropertyPath(propertySelector)
    ? ObserveNestedProperty(subject, propertySelector, isPushCurrentValueAtFirst)
    : ObserveSimpleProperty(subject, propertySelector, isPushCurrentValueAtFirst);
```

File: Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs [hunk 3]
```diff
@@ -92,15 +118,27 @@ public static IObservable<PropertyChangedEventArgs> PropertyChangedAsObservable<
...
-            var setter = AccessorCache<TSubject>.LookupSet(propertySelector, out _);
-
-            var result = subject.ObserveProperty(propertySelector, isPushCurrentValueAtFirst: true)
-                .ToReactiveProperty(raiseEventScheduler, mode: mode);
-            result
-                .Where(_ => !ignoreValidationErrorValue || !result.HasErrors)
-                .Subscribe(x => setter(subject, x));
-
-            return result;
+            if (!ExpressionTreeUtils.IsNestedPropertyPath(propertySelector))
```
Comment (xin9le) [inline Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs:121]: It's the same here.
It's easier to understand if swap `if` and `else`.

File: Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs (hunk 2 above)
Comment (runceel) [reply Source/ReactiveProperty.NETStandard/Extensions/INotifyPropertyChangedExtensions.cs]: Thanks! I swapped if and else!

**Comment by xin9le** [comment]
image

**Pull request merged**